### PR TITLE
Developent

### PR DESCRIPTION
**Pull request by mfiguera**
Project is now compiling and solving smaller puzzles. First "working" version.

### CODE REVIEW

File: includes/nodes.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+/*   Created: 2019/12/11 20:43:55 by jpinyot           #+#    #+#             */
+/*   Updated: 2020/09/28 11:15:12 by mfiguera         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "state.h"
+
+/* if(currentState.canMoveTo(N)) { */
+
+/* } */
```
Comment (mfigueras11) [inline includes/nodes.h:17]: Clean this

File: includes/state.h [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2020/09/28 11:14:12 by mfiguera          #+#    #+#             */
+/*   Updated: 2020/10/01 09:33:22 by jpinyot          ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#pragma once
+#include <vector>			//vector 
+#include <list>
+#include <unistd.h>			//write
```
Comment (mfiguera) [inline includes/state.h:16]: Is <unistd.h> necessary?

File: includes/closedStack.h [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+class ClosedStack
+{
+	private:
+		vector<State *>	stack_;
+	public:
+		ClosedStack():
+			stack_(k_maxScore, NULL)
+		{;}
+
+		bool	stateIsClosed(State* state);
```
Comment (jpinot) [inline includes/closedStack.h]: Need to be defined as const??

**Review by mfiguera — COMMENTED** [review]
Need to clean some comments.

File: includes/openStack.h [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        {
+            top_ = NULL;
+        }
+
+        void    addState(State *state);
+        State   *getTop() {return top_;}
+        
+        void    popTop();
+        void    display();
+        bool    stateIsOpen(State *state);
```
Comment (jpinot) [inline includes/openStack.h]: Need to be defined as const?

File: includes/state.h (hunk 2 above)
Comment (jpinot) [reply includes/state.h:16]: Useful for debugging.

File: includes/state.h [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+	const bool	isSolved() {return (heuristicScore_ == 0);}
+	/* void		betterScore(const State* currState); */
+
+	const bool	canMoveTo(const Moves& move);
+		void		display();
+        virtual const int       singleTileDistance(int tile) = 0;
+
+};
+
+/* list<int, vector<State*> > visited; */
```
Comment (jpinot) [inline includes/state.h:86]: Delete line.

File: includes/stateManhattanDistance.h [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+	    	setHeuristicScoreFromPrev();
+    		/* display(); */
+
+	    };
+
+        StateManhattanDistance(const vector<unsigned char> puzzle, const Moves& move=none) :
+		State(puzzle, move)
+        {
+            puzzle_ = puzzle;
+            //itPos_ need to be initialized
```
Comment (jpinot) [inline includes/stateManhattanDistance.h]: set as TODO if is not do it yet, delete line if already done

**Review by jpinot — CHANGES_REQUESTED** [review]
Need change on some files, set methods const when needed!

**Review by jpinot — APPROVED** [review]
WIP branch. All request fixed.

**Pull request merged**